### PR TITLE
onPaste 

### PR DESCRIPTION
**Pull request by sorja**
note: table 2a onPaste doesn't work. to investigate

### CODE REVIEW

**Comment by minotogna** [comment]
please give a better name to this PR

File: meta/assessment/cols.ts [hunk 1]
```diff
@@ -22,7 +22,20 @@ const getColName = (props: { colIdx: number; cols: Array<Col> }): string => {
   return col.props.colName ?? '' // `"${col.props.colName ?? ''}"`
 }
 
+const isReadOnly = (props: { col: Col; row: Row }): boolean => {
+  const { col, row } = props
+  if (
+    row.props.readonly ||
+    row.props.calculateFn ||
+    col.props.calculateFn ||
```
Comment (sorja) [inline meta/assessment/cols.ts:30]: col.props.odp ?

File: client/store/pages/assessmentSection/slice.ts [hunk 2]
```diff
@@ -42,14 +43,17 @@ export const assessmentSectionSlice = createSlice({
...
-    builder.addCase(updateNodeValue.fulfilled, (state, { payload }) => {
-      const { colName, countryIso, tableName, variableName, value } = payload
-      state.data = TableDatas.updateDatum({ colName, countryIso, tableName, data: state.data, variableName, value })
-    })
-
     builder.addCase(getOriginalDataPointData.fulfilled, (state, { payload }) => {
       state.originalDataPointData = payload
     })
+
+    builder.addCase(updateNodeValues.pending, (state, { meta }) => {
```
Comment (sorja) [inline client/store/pages/assessmentSection/slice.ts:50]: why pending? what if server says no? inconsistent state then?

File: meta/assessment/cols.ts [hunk 3]
```diff
@@ -22,7 +22,20 @@ const getColName = (props: { colIdx: number; cols: Array<Col> }): string => {
   return col.props.colName ?? '' // `"${col.props.colName ?? ''}"`
 }
 
+const isReadOnly = (props: { col: Col; row: Row }): boolean => {
```
Comment (sorja) [inline meta/assessment/cols.ts:25]: should we have odp information here?

Comment (minotogna) [reply meta/assessment/cols.ts:25]: that's at col level.

**Review by minotogna — APPROVED** [review]

**Pull request merged**